web/src/main/java/com/graphhopper/http/GraphHopperServlet.java

```diff
@@ -90,6 +90,13 @@ public void doGet(HttpServletRequest httpReq, HttpServletResponse httpRes) throw
                     throw new IllegalArgumentException("heading list in from format: " + e.getMessage());
                 }
 
+                List<String> pointHints = Collections.EMPTY_LIST;
+                pointHints = new ArrayList<String>(Arrays.asList(getParams(httpReq, "point_hint")));
+                if (pointHints.size() > 0 && pointHints.size() != requestPoints.size()) {
+                    // TODO: Not sure if this is nice?
+                    throw new IllegalArgumentException("If you pass point_hints, you need to pass a hint for every point, empty hints will be ignored");
```

Comment:
This is good! We do similar things for heading I think. Minor: `point_hints` -> `point_hint`